lib/job-iteration/active_storage_csv_enumerator.rb

```diff
@@ -0,0 +1,98 @@
+# froze_string_literal: true
+
+module JobIteration
+    class ActiveStorageCsvEnumerator
+        
+        """
+        @param [ActiveStorage::Attachment, ActiveStorage::Blob] attachment_or_blob
+        @param [Integer] cursor - byte offset to start reading from
+        @param [Boolean] include_header
+        @param [Integer] batch_size - # of rows to read
+        @param [Integer] chunk_size - # of bytes to read at a time
+        @param [Hash] parse_opts - options to pass to CSV
+        """
+        def initialize(attachment_or_blob, cursor: nil, include_header: false, batch_size: 100, chunk_size: 256, **parse_opts)
+            @blob = if attachment_or_blob.respond_to?(:blob)
+                attachment_or_blob.blob
+            else
+                attachment_or_blob
+            end
+
+            @cursor = cursor || 0
+            @batch_size = batch_size
+            @chunk_size = chunk_size
+            @parse_opts = parse_opts
+
+            @row_sep = @parse_opts[:row_sep] || "\n"
+            
+            # read header
+            hdr, hdr_cursor = ingest_row(0)
+            @cursor = hdr_cursor if hdr_cursor > @cursor
+            @parse_opts[:headers] = CSV.parse(hdr)[0] if include_header
+        end
+
+        def rows
+            Enumerator.new() do |yielder|
```

Comment:
Not strictly necessary but easy enough to add and nice to be able to call `size` on the enumerator?
```suggestion
            Enumerator.new(@blob.byte_size) do |yielder|
```
Same for the batches method below.